components/ntp_sponsored_images/browser/ntp_sponsored_images_service.h

```diff
@@ -10,35 +10,39 @@
 #include <string>
 
 #include "base/files/file_path.h"
+#include "base/gtest_prod_util.h"
 #include "base/memory/weak_ptr.h"
 #include "base/observer_list.h"
-#include "base/optional.h"
-#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_image_source.h"
-#include "components/component_updater/component_updater_service.h"
 
-struct NTPSponsoredImagesData;
-struct NTPSponsoredImagesInternalData;
+FORWARD_DECLARE_TEST(NTPSponsoredImagesServiceTest, InternalDataTest);
+
+namespace component_updater {
+class ComponentUpdateService;
+}
 
 namespace content {
 class BrowserContext;
 }  // namespace content
 
-// NOTE: Client should call AddDataSource() before requesting image sources.
-// When component is updated, Client also should call AddDataSource() for add
-// proper |URLDataSource|s.
-class NTPSponsoredImagesService : public component_updater::ServiceObserver {
```

Comment:
@bridiver why this inheritance was deleted?
W/o this, we can't know component update during the runtime.

Reply:
NVM! I confirmed that `ComponentInstallerPolicy::ComponentReady()` is called. 👍